Correct coordinate calculation for rectangle

vanicat opened the pull request:
The new computation
- scale the rectangle as anyother coordinate
- use self.tiled_map.map_size.height * self.tiled_map.tile_size[1] to offset the y coordinate, otherwise they are negative.

Fix  #1740

pvcraven commented:
Looks like it didn't pass tests? Please research if you could.

gran4 reviewed (changes requested):
Re-Add the 2 lines(921 and 922):
`x = cur_object.coordinates.x + offset[0]`
and
`y = cur_object.coordinates.y + offset[1]`
You deleted. Without them, x and y don't exist.

vanicat commented:
I will look at the test after the end of Ludumdare,

vanicat committed:
The new computation
- scale the rectangle as anyother coordinate
- use self.tiled_map.map_size.height * self.tiled_map.tile_size[1] to offset the y coordinate,
otherwise they are negative.

vanicat commented:
I've updated the patch, and it now pass all the test.
The two line proposed by gran4 was not correct, because the offset must be apply after the scaling (at least, it is what happen for others coordinate computation).

Thanks.

gran4 commented:
> I've updated the patch, and it now pass all the test. The two line proposed by gran4 was not correct, because the offset must be apply after the scaling (at least, it is what happen for others coordinate computation).

I didn't mean to fix the bug, I meant to get it to run.

vanicat committed:
Add new test for tiled.

Some computation are made to translate the coordinate système of tiled
to the one of arcade. Those 3 new test check they are ok.
